mesheryctl/internal/cli/root/registry/generate.go

```diff
@@ -112,22 +114,50 @@ mesheryctl registry generate --spreadsheet-id "1DZHnzxYWOlJ69Oguz4LkRVTFM79kC2tu
 		}
 		var err error
 
-		srv, err = mutils.NewSheetSRV(spreadsheeetCred)
+		// isCsvPathPresent := modelCSVFilePath != "" && componentCSVFilePath != ""
+		if csvDirectory == "" {
+			srv, err = mutils.NewSheetSRV(spreadsheeetCred)
+			if err != nil {
+				utils.LogError.Error(ErrUpdateRegistry(err, modelLocation))
+				return err
+			}
+
+			resp, err := srv.Spreadsheets.Get(spreadsheeetID).Fields().Do()
+			if err != nil || resp.HTTPStatusCode != 200 {
+				utils.LogError.Error(ErrUpdateRegistry(err, outputLocation))
+				return err
+			}
+
+			// Collect list of Models by name from spreadsheet
+			sheetGID = GetSheetIDFromTitle(resp, "Models")
+			// Collect list of corresponding Components by name from spreadsheet
+			componentSpredsheetGID = GetSheetIDFromTitle(resp, "Components")
+		}
+		// Get all files in the directory
+		files, err := os.ReadDir(csvDirectory)
 		if err != nil {
-			utils.LogError.Error(ErrUpdateRegistry(err, modelLocation))
-			return err
+			return fmt.Errorf("error reading the directory: %v", err)
 		}
+		for _, file := range files {
+			filePath := filepath.Join(csvDirectory, file.Name())
+			if !file.IsDir() && strings.HasSuffix(file.Name(), ".csv") {
+				headers, secondRow, err := getCSVHeader(filePath)
+				if Contains(headers, "modelDisplayName") || Contains(secondRow, "modelDisplayName") {
+					modelCSVFilePath = filePath
+				} else if Contains(headers, "component") || Contains(secondRow, "component") { // Check if the file matches the ComponentCSV structure
+					componentCSVFilePath = filePath
+					continue
+				}
+				if err != nil {
+					return fmt.Errorf("error checking file %s: %v", file.Name(), err)
+				}
 
-		resp, err := srv.Spreadsheets.Get(spreadsheeetID).Fields().Do()
-		if err != nil || resp.HTTPStatusCode != 200 {
-			utils.LogError.Error(ErrUpdateRegistry(err, outputLocation))
-			return err
+			}
```

Comment:
This will run even though I provide the sheet Id and not the csv? Should this be in else?